cpp/ql/test/library-tests/files/Files.ql

```diff
@@ -0,0 +1,18 @@
+import cpp
+
+string describe(File f) {
+  f.compiledAsC() and
+  result = "C"
+  or
+  f.compiledAsCpp() and
+  result = "C++"
+  or
+  f instanceof XMLParent and
+  result = "XMLParent" // regression tests a bug in the constructor of XMLParent
```

Comment:
```suggestion
  result = "XMLParent" // regression tests a bug in the characteristic predicate of XMLParent
```

Reply:
This case doesn't actually seem to be exercised by the test?

Reply:
It is in the second commit, then the third (rightly) removed those results.  So if we ever regressed the char pred of XMLParent, those results would show up again.

Reply:
Makes sense.